lib/reqTemplate.js

```diff
@@ -230,44 +212,63 @@ function createURIResolver(spec) {
  *              fields that couldn't be resolved from original request would be ignored.
  */
 function Template(spec) {
+    spec = Object.assign({}, spec);
     var self = this;
-    self.resolvers = [];
-    if (typeof spec === 'string') {
-        self.resolvers.push(compileTemplate(spec, function(newReq, val) {
-            newReq.val = val;
-        }));
-
-        self._eval = function(context) {
-            var result = {};
-            this.resolvers.forEach(function(resolver) {
-                resolver(result, context);
-            });
-            return result.val;
-        };
-    } else {
-        Object.keys(spec).forEach(function(reqPart) {
-            var setValue;
-            if (reqPart === 'uri') {
-                self.resolvers.push(createURIResolver(spec));
-            } else if (reqPart === 'method') {
-                setValue = _setAtPath('method', spec);
-                self.resolvers.push(function(newReq) {
-                    setValue(newReq, spec.method);
-                });
-            } else if (spec[reqPart]) {
-                self.resolvers = self.resolvers.concat(
-                    _createTemplateResolvers(spec, spec[reqPart], reqPart));
-            }
-        });
+    var globals = Object.assign({}, globalMethods);
+    globals._i = 0;
+    Object.keys(spec).forEach(function(part) {
+        if (part === 'uri') {
+            globals.uri = createURIResolver(spec);
+            spec.uri = '$$.uri($context)';
+        } else if (part === 'method') {
+            spec.method = "'" + (spec.method || 'get') + "'";
+        } else {
+            spec[part] = replaceComplexTemplates(part, spec[part], globals);
+        }
+    });
 
-        self._eval = function(context) {
-            var newReq = { method: context.request.method };
-            this.resolvers.forEach(function(resolver) {
-                resolver(newReq, context);
-            });
-            return newReq;
-        };
+    var completeTAssemblyTemplate;
+    try {
+        completeTAssemblyTemplate = expressionCompiler.parse(spec);
+    } catch (e) {
+        console.log('COMPILE ERROR');
+        console.log(JSON.stringify(spec, null, 2));
+        console.log(e);
+        throw e;
     }
+
+    var res;
+    var callback = function(bit) {
+        if (res === undefined) {
+            res = bit;
+        } else {
+            res += '' + bit;
+        }
+    };
+    var resolver = TAssembly.compile([['raw', completeTAssemblyTemplate]], {
```

Comment:
Yeah, that's definitely worth looking into. I mostly went for the `nesteTemplate` mode for a little bit extra performance, but you are right that it might not be worth it.